### PR TITLE
fix: Telegram latest message should always return newest turn

### LINKED ISSUE #367: Fix Telegram Latest message returning oldest turn
## Context
Telegram's **Latest message** inline button (shown on task-finished notifications) still often returns the first/oldest turn instead of the true latest turn. This diverges from web UI behavior, where users reliably see the newest turn.

Current implementation is in `shared/telegram-bridge.ts`:
- callback handler uses `recentText(..., 1, ..., { preserveFullLatest: true })`
- `recentText` fetches `/session/{id}/message?limit=...` and then re-sorts locally by `created` or fallback `id`

This can produce incorrect ordering when timestamps are missing/inconsistent or when API ordering + limit do not represent full chronological order.

## Goal
Make Telegram "Latest message" deterministic and correct: always return the newest user+assistant exchange for the targeted session.

## Scope / Files
- `shared/telegram-bridge.ts`
- `app-prefixable/tests/telegram-bridge.test.ts`

## Proposed Direction
- Avoid selecting "latest" from a potentially head-limited/reordered subset.
- For latest-turn lookup, prefer a robust strategy that preserves server chronology and only falls back safely when metadata is missing.
- Ensure the button continues to target the originating session token (`u:r:*`) and does not drift to another mapped session.
- Keep full assistant content for latest lookup; if response exceeds Telegram single-message limits, rely on existing chunked send behavior to emit multiple messages.

## Acceptance Criteria
- Pressing **Latest message** after task-finished notification returns the newest turn from the originating session.
- Result never regresses to first/oldest turn when timestamps are absent or non-monotonic.
- Long latest replies are delivered fully (single or multiple Telegram messages as needed), not cut to preview length.
- Existing behavior for `/recent [count]` remains intact unless explicitly updated by follow-up work.
- Tests cover ordering edge cases (missing/duplicate timestamps, non-chronological IDs) and pass.

### PR DESCRIPTION
**Pull request by geier**
Closes #367

### CODE REVIEW

File: shared/telegram-bridge.ts [hunk 1]
```diff
@@ -1928,7 +1965,9 @@ async function recentText(
   const safeCount = Math.max(1, Math.min(count, recentMaxCount))
   const keepFull = options?.preserveFullLatest === true && safeCount === 1
   const url = opencodeUrl(config, `/session/${encodeURIComponent(sessionId)}/message`)
-  url.searchParams.set("limit", String(Math.max(20, safeCount * 6)))
+  if (!keepFull) {
+    url.searchParams.set("limit", String(Math.max(20, safeCount * 6)))
+  }
```
[the remark this answers is not in the document]
Comment (geier) [reply shared/telegram-bridge.ts:1970]: Thanks — agreed on the trade-off. For this issue we intentionally avoid `limit` in `preserveFullLatest` mode because bounded oldest-first responses can miss the true latest turn and recreate the bug. I kept the correctness-first behavior here and can open a follow-up optimization issue for a two-step retrieval approach (metadata selection + full fetch for chosen turn).

File: app-prefixable/tests/telegram-bridge.test.ts [hunk 2]
```diff
@@ -7463,6 +7463,106 @@ describe("telegram bridge config and cache", () => {
...
+    const latest = calls
+      .filter((x) => x.url.includes("/sendMessage"))
+      .map((x) => String(x.body.text || ""))
+      .find((text) => text.includes("Recent activity for session")) || "";
+    expect(latest).toContain("You: Newest question");
+    expect(latest).toContain("Assistant: Newest answer");
+    expect(latest).not.toContain("Older question");
+
+    const messageUrl = calls.find((x) => x.url.includes("/session/session-2/message"))?.url || "";
+    expect(messageUrl.includes("limit=")).toBe(false);
```
[the remark this answers is not in the document]
Comment (geier) [reply app-prefixable/tests/telegram-bridge.test.ts]: Good catch — updated the test to assert across all matching `/session/session-2/message` requests so none include `limit=`. This is now covered by filtering all matching calls and checking each query string.
